src/utils/errors.ts

```diff
@@ -540,3 +540,20 @@ export class ShortLinkLimitExceed extends CoreError {
     this.origin = 'shortLink';
   }
 }
+
+export class ItemTypeNotAllowedToPublish extends CoreError {
+  constructor(itemId: string, itemType: ValuesOfItemType) {
+    super(
+      {
+        code: 'GERR1012',
+        statusCode: StatusCodes.BAD_REQUEST,
+        message: `The type "${itemType}" of item "${itemId}" is not allowed to be published. Only theses types are allowed "${ALLOWED_PUBLISHED_TYPES}".`,
+      },
+      {
+        itemId,
+        itemType,
+      },
+    );
+    this.origin = 'published';
+  }
+}
```

Comment:
Actually there's a file `errors.ts` in published where you can save this new error.

Reply:
Nice, thanks !